core/src/main/java/com/google/errorprone/bugpatterns/StatementSwitchToExpressionSwitch.java

```diff
@@ -282,11 +293,37 @@ private static AnalysisResult analyzeSwitchTree(SwitchTree switchTree, VisitorSt
             // The switch must be exhaustive (at compile time)
             && exhaustive;
 
+    List<StatementTree> precedingStatements = getPrecedingStatementsInBlock(switchTree, state);
+    Optional<VariableTree> variableTreeOptional =
+        precedingTwoStatementsMultiVariableDeclarations(precedingStatements, state)
+            ? Optional.empty()
+            : (precedingStatements.isEmpty()
+                ? Optional.empty()
+                : Optional.of(precedingStatements.get(precedingStatements.size() - 1))
+                    .filter(precedingStatement -> precedingStatement instanceof VariableTree)
+                    .map(precedingStatement -> (VariableTree) precedingStatement)
+                    .filter(
+                        variableTree ->
+                            (variableTree.getInitializer() == null)
+                                || (compileTimeConstExpressionMatcher.matches(
+                                        variableTree.getInitializer(), state))
+                                    && noReferences(ASTHelpers.getSymbol(variableTree), state)));
+
+    boolean compatibleVariableDeclaration =
+        canConvertToAssignmentSwitch
+            && variableTreeOptional.isPresent()
+            && isCompatibleWithFirstAssignment(
+                assignmentSwitchAnalysisState.assignmentTargetOptional(),
+                variableTreeOptional.get());
+
+    System.out.println("XXX compatibleVariableDeclaration: " + compatibleVariableDeclaration);
```

Comment:
Likely to also be caught by Google-internal code review, but: I assume these stdout print statements are meant to be dropped before the PR is merged :eyes: